processor/sumologicprocessor/processor.go

```diff
@@ -0,0 +1,114 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sumologicprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/sumologicprocessor"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/processor"
+	"go.uber.org/zap"
+)
+
+type sumologicSubprocessor interface {
+	processLogs(plog.Logs) error
+	processMetrics(pmetric.Metrics) error
+	processTraces(ptrace.Traces) error
+	isEnabled() bool
+	ConfigPropertyName() string
+}
+
+type sumologicProcessor struct {
+	logger        *zap.Logger
+	subprocessors []sumologicSubprocessor
+}
+
+func newsumologicProcessor(set processor.CreateSettings, config *Config) *sumologicProcessor {
+	cloudNamespaceProcessor := newCloudNamespaceProcessor(config.AddCloudNamespace)
+
+	translateAttributesProcessor := newTranslateAttributesProcessor(config.TranslateAttributes)
+
+	translateTelegrafMetricsProcessor := newTranslateTelegrafMetricsProcessor(config.TranslateTelegrafAttributes)
+
+	nestingProcessor := newNestingProcessor(config.NestAttributes)
+
+	aggregateAttributesProcessor := newAggregateAttributesProcessor(config.AggregateAttributes)
+
+	logFieldsConversionProcessor := newLogFieldConversionProcessor(config.LogFieldsAttributes)
+
+	translateDockerMetricsProcessor := newTranslateDockerMetricsProcessor(config.TranslateDockerMetrics)
+
+	processors := []sumologicSubprocessor{
+		cloudNamespaceProcessor,
+		translateAttributesProcessor,
+		translateTelegrafMetricsProcessor,
+		nestingProcessor,
+		aggregateAttributesProcessor,
+		logFieldsConversionProcessor,
+		translateDockerMetricsProcessor,
+	}
+
+	processor := &sumologicProcessor{
+		logger:        set.Logger,
+		subprocessors: processors,
+	}
+
+	return processor
+}
+
+func (processor *sumologicProcessor) start(_ context.Context, _ component.Host) error {
+	procs := processor.subprocessors
+	processor.logger.Info(
+		"Sumo Logic Processor has started.",
+		zap.Bool(procs[0].ConfigPropertyName(), procs[0].isEnabled()),
+		zap.Bool(procs[1].ConfigPropertyName(), procs[1].isEnabled()),
+		zap.Bool(procs[2].ConfigPropertyName(), procs[2].isEnabled()),
+		zap.Bool(procs[3].ConfigPropertyName(), procs[3].isEnabled()),
+		zap.Bool(procs[4].ConfigPropertyName(), procs[4].isEnabled()),
+		zap.Bool(procs[5].ConfigPropertyName(), procs[5].isEnabled()),
```

Comment:
I believe you could convert `procs` to a `[]zap.Bool` and then spread it here, something like: `logger.Info("", enabledProcessors...)`.

Reply:
Converted :)